src/main/java/bio/terra/service/filedata/azure/util/BlobCrl.java

```diff
@@ -109,8 +114,35 @@ public BlobContainerCopier createBlobContainerCopier(URL sourceUrl, String desti
    *
    * @param blobName blob name to delete.
    */
-  public void deleteBlob(String blobName) {
-    blobContainerClientFactory.getBlobContainerClient().getBlobClient(blobName).delete();
+  public boolean deleteBlob(String blobName) {
+    try {
+      blobContainerClientFactory.getBlobContainerClient().getBlobClient(blobName).delete();
+      return true;
+    } catch (BlobStorageException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+        return false;
+      } else {
+        throw new PdaoException("Error deleting file", e);
```

Comment:
After looking at DR-2030 and reading the description I'm not sure if this is what is intended by adding a boolean result. Right now it will only return `false` if the blob can't be deleted if it doesn't exist; all other cases result in a thrown exception. From reading the description it sounds like the code should swallow any exception and callers need to examine the boolean result to determine if an error occurred or not.

I guess an overall question is, what bug in our code is fixed by this PR? Is there some code that failed before and would work successfully now?

Reply:
I think this is solely a reorganization so that we don't have to repeat the try/catch logic in multiple places.

Reply:
Gotcha, that makes sense. I'll take a look at how this API is currently used, then.